detox/ios/Detox/Invocation/WKWebViewConfiguration+Detox.m

```diff
@@ -16,61 +28,130 @@ @interface DTXFakeWKPreferencesRef: NSObject
 @implementation DTXFakeWKPreferencesRef
 @end
 
+
 /// Set web-security policy for WebKit (e.g. CORS restriction).
 ///
 /// @note Since we can't access the `WKPreferencesSetWebSecurityEnabled` directly with
-///  a `WKPreferences*`, we wrap it in a `WKPreferencesRef`, which can be passed to this function.
+///  a `WKPreferences*`, we wrap it in a `WKPreferencesRef` in the getWKPrefsRef function, which can be passed to this function.
 /// This private API is not officially supported on iOS, and generally used for debugging / testing
 ///  purposes on MacOS only. So there's no guarantee that it will work in the future.
 void DTXPreferencesSetWebSecurityEnabled(WKPreferences* prefs, bool enabled) {
-	DTXFakeWKPreferencesRef* fakeRef = [DTXFakeWKPreferencesRef new];
-
-	Ivar ivar = class_getInstanceVariable([WKPreferences class], "_preferences");
-	void* realPreferences = (void*)(((uintptr_t)prefs) + ivar_getOffset(ivar));
-	fakeRef._apiObject = realPreferences;
-
-	WKPreferencesSetWebSecurityEnabled(fakeRef, enabled);
+    Ivar ivar = NULL;
+    
+    if (@available(iOS 18.0, *)) {
+        unsigned int ivarCount;
+        Ivar *ivars = class_copyIvarList([WKPreferences class], &ivarCount);
+        if (ivars) {
+            for (unsigned int i = 0; i < ivarCount; i++) {
+                const char *ivarName = ivar_getName(ivars[i]);
+                if (ivarName && strcmp(ivarName, "_preferences") == 0) {
+                    ivar = ivars[i];
+                    break;
+                }
+            }
+            free(ivars);
```

Comment:
- not a fan of `break`, unless commonly used, please avoid using it; is there a functional-programming alternative to this `for` loop? (e.g. array.some())
- `ivars` and `ivar` is really close and a bit confusing

other than that - looks ok 😄